src/main/java/com/auction/auction/bid/controller/UserController.java

```diff
@@ -1,52 +1,46 @@
 package com.auction.auction.bid.controller;
 
 import com.auction.auction.bid.dto.UserDTO;
+import com.auction.auction.bid.interfaces.UserInterface;
 import com.auction.auction.bid.model.User;
 import com.auction.auction.bid.service.UserService;
 import jakarta.validation.Valid;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.*;
-
 import java.util.List;
 
 @RestController
 @RequestMapping("/users")
 public class UserController {
 
     @Autowired
-    private final UserService userService;
+    private UserInterface userService;
 
-    public UserController(UserService userService) {
-        this.userService = userService;
-    }
 
     @PostMapping("/users")
     public ResponseEntity<User> createUser(@RequestBody @Valid UserDTO userDTO) {
-        User user = new User(
-                userDTO.getUsername(),
-                userDTO.getPassword(),
-                userDTO.getEmail()
-        );
-        User createdUser = userService.createUser(user);
-        return ResponseEntity.status(HttpStatus.CREATED).body(createdUser);
+        User response = userService.createUser(userDTO);
+        return ResponseEntity.status(HttpStatus.CREATED).body(response);
+    }
+
+    @PutMapping("/{id}")
+    public ResponseEntity<User> updateUser(@PathVariable String id, @RequestBody User user) {
+        User updatedUser = userService.updateUser(id, user);
+        return new ResponseEntity<>(updatedUser, HttpStatus.OK);
     }
 
-    @PostMapping("/saveOrUpdate")
-    public ResponseEntity<User> saveOrUpdateUser(@RequestBody User user) {
-        User savedUser = userService.saveOrUpdateUser(user);
+    @PostMapping("/save")
```

Comment:
you should not use "save" because if we are using PostMapping in UserController - it is clear that it will be User creation even without additional information in mapping